src/test/rpc/LedgerRPC_test.cpp

```diff
@@ -0,0 +1,288 @@
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/protocol/ErrorCodes.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/test/jtx.h>
+#include <ripple/beast/unit_test.h>
+
+namespace ripple {
+
+class LedgerRPC_test : public beast::unit_test::suite
+{
+    static
+    std::unique_ptr<Config>
+    makeNonAdminConfig()
+    {
+        auto p = std::make_unique<Config>();
+        test::setupConfigForUnitTests(*p);
+        (*p)["port_rpc"].set("admin","");
+        (*p)["port_ws"].set("admin","");
+        return p;
+    }
+
+public:
+    void testBadInput()
+    {
+        testcase("Bad Input");
+        using namespace test::jtx;
+        Env env { *this };
+        Account const gw { "gateway" };
+        auto const USD = gw["USD"];
+        Account const bob { "bob" };
+
+        env.fund(XRP(10000), gw, bob);
+        env.trust(USD(1000), bob);
+        env.close();
+
+        {
+            Json::Value jvParams;
+            jvParams[jss::ledger_index] = "0"; // NOT an integer
+            auto const jrr = env.rpc ( "json", "ledger", to_string(jvParams) ) [jss::result];
+            BEAST_EXPECT(jrr[jss::error]         == "invalidParams");
+            BEAST_EXPECT(jrr[jss::status]        == "error");
+            BEAST_EXPECT(jrr[jss::error_message] == "ledgerIndexMalformed");
+        }
+
+        {
+            // ask for a bad ledger index
+            Json::Value jvParams;
+            jvParams[jss::ledger_index] = 10u;
+            auto const jrr = env.rpc ( "json", "ledger", to_string(jvParams) ) [jss::result];
+            BEAST_EXPECT(jrr[jss::error]         == "lgrNotFound");
+            BEAST_EXPECT(jrr[jss::status]        == "error");
+            BEAST_EXPECT(jrr[jss::error_message] == "ledgerNotFound");
+        }
+
+        {
+            // arbitrary text is converted to 0.
+            auto const jrr = env.rpc("ledger", "arbitrary_text") [jss::result];
+            BEAST_EXPECT(jrr[jss::error]         == "lgrNotFound");
+            BEAST_EXPECT(jrr[jss::status]        == "error");
+            BEAST_EXPECT(jrr[jss::error_message] == "ledgerNotFound");
+        }
+
+    }
+
+    void testLedgerRequest()
+    {
+        testcase("Basic Request");
+        using namespace test::jtx;
+
+        Env env {*this};
+
+        env.close();
+        env.close();
```

Comment:
that particular (anti?)-pattern was taken from this test:

https://github.com/ripple/rippled/blob/faac234f7eeeee4761dcedc72fcfd0657117e3d7/src/test/rpc/LedgerRequestRPC_test.cpp#L53

...and I *think* it's just doing it to advance the ledger a few times. I don't honestly see how it adds any value in these tests, so I will remove this extra close - thanks for pointing that out.